lib/bunch/binary.ex

```diff
@@ -5,6 +5,74 @@ defmodule Bunch.Binary do
 
   use Bunch
 
+  @doc """
+  Returns a tuple `{list, rest}` where each element of the list is the result of
+  invoking `f` on a part of binary, and rest is the unprocessed leftover.
+
+  Function `f` is always given entire non-consumed part binary and is supposed to
+  return a subsequent output element and the remaining part of the binary to be
+  processed by further calls to the function. Once the remaining data becomes
+  ineligible for processing, the function should return `:halt`.
```

Comment:
How long have you been going through thesaurus to find the word `ineligible`? Besides, what if data is still eligible for processing but you want to stop it either way?
Please make it simpler.

Reply:
Something like:
```markdown
Function `f` gets a non-consumed part of the binary and should return either a tuple with
processing (matching?) result and the rest of the binary or an atom `:halt`.
```